pkg/agent/flowexporter/connections/conntrack_connections.go

```diff
@@ -106,7 +107,10 @@ func (cs *ConntrackConnectionStore) Poll() ([]int, error) {
 	klog.V(2).Infof("Polling conntrack")
 	// DeepCopy the L7EventMap before polling the conntrack table to match corresponding L4 connection with L7 events
 	// and avoid missing the L7 events for corresponding L4 connection
-	l7EventMap := cs.l7EventMapGetter.ConsumeL7EventMap()
+	var l7EventMap map[flowexporter.ConnectionKey]L7ProtocolFields
+	if !reflect.ValueOf(cs.l7EventMapGetter).IsNil() {
```

Comment:
if we have to use reflection to check this something is wrong. `cs.l7EventMapGetter` should be a nil interface if the feature is disabled

Reply:
I've modified it to set it to nil if l7Listener is nil (L7FlowExporter is disabled) when creating a NewConntrackConnectionStore.